store/sql_oauth_store.go

```diff
@@ -34,21 +36,36 @@ func NewSqlOAuthStore(sqlStore *SqlStore) OAuthStore {
 		tableAuth.ColMap("Scope").SetMaxSize(128)
 
 		tableAccess := db.AddTableWithName(model.AccessData{}, "OAuthAccessData").SetKeys(false, "Token")
-		tableAccess.ColMap("AuthCode").SetMaxSize(128)
+		tableAccess.ColMap("ClientId").SetMaxSize(26)
+		tableAccess.ColMap("UserId").SetMaxSize(26)
 		tableAccess.ColMap("Token").SetMaxSize(26)
 		tableAccess.ColMap("RefreshToken").SetMaxSize(26)
 		tableAccess.ColMap("RedirectUri").SetMaxSize(256)
+		tableAccess.SetUniqueTogether("ClientId", "UserId")
 	}
 
 	return as
 }
 
 func (as SqlOAuthStore) UpgradeSchemaIfNeeded() {
+	as.CreateColumnIfNotExists("OAuthApps", "IsTrusted", "tinyint(1)", "boolean", "0")
+	as.CreateColumnIfNotExists("OAuthApps", "IconURL", "varchar(512)", "varchar(512)", "")
+	as.CreateColumnIfNotExists("OAuthAccessData", "ClientId", "varchar(26)", "varchar(26)", "")
+	as.CreateColumnIfNotExists("OAuthAccessData", "UserId", "varchar(26)", "varchar(26)", "")
+	as.CreateColumnIfNotExists("OAuthAccessData", "ExpiresAt", "bigint", "bigint", "0")
+
+	// ADDED for 3.3 REMOVE for 3.7
+	if as.DoesColumnExist("OAuthAccessData", "AuthCode") {
+		as.GetMaster().Exec("ALTER TABLE OAuthAccessData DROP INDEX idx_oauthaccessdata_auth_code")
```

Comment:
Does this work for both mysql and postgress.  I wonder is we should use `RemoveIndexIfExists()` in sql_store.go